docs/performance/inserts/tuning.txt

```diff
@@ -0,0 +1,141 @@
+
+Use the ``iostat`` tool to display disk I/O statistics, which should include
+output like this:
+
+.. code-block:: text
+
+    sh$ iostat
+    ...
+    avg-cpu:  %user   %nice %system %iowait  %steal   %idle
+               0.53    0.01    0.05    0.01    0.00   99.41
+    ...
+
+Here, the most interesting metric is ``%iowait``. This tells you how
+time the CPU is waiting for disk I/O. You can expect better insert performance
+from CrateDB relative to how low this value is.
+
+CPUs
+----
+
+If your table schemas are complex or make use of `fulltext indexes`_, the
+additional CPU overhead during the analyzing phase of insert query processing
+might result in a CPU performance bottleneck.
+
+If this becomes the case, you might want to try using different fulltext
+analyzers or maybe even turn off fulltext indexing entirely. If neither of
+these are an option, upgrading your CPUs will be a cost-effective way to boost
+CrateDB performance.
+
+Software
+========
+
+Replicas
+--------
+
+The biggest performance cost occurred when you go from zero replicas to one
+configured replica for a table. After that point, load increases linearly. This
+is because a write is made to the primary first, and then afterwards, writes
+are made concurrently to the replicas.
+
+However, using tables with zero replicas is not recommended for anything except
+one-off data imports. You should have at least one configured replica for every
+table in a cluster that has three nodes or more.
+
+Replicas improve availability and durability (in the event of node failure or
+cluster partitions) but they do incur a performance cost.
+
+Shards and Partitioning
+-----------------------
+
+The :ref:`sharding guide <sharding_guide>` has a section that covers
+:ref:`ingestion performance <sharding_ingestion>`.
+
+Indexing
+--------
+
+By default, all table columns are indexed. Regular columns use the ``plain``
+index, and fulltext columns use the ``fulltext`` index.
+
+Indexes are expensive, so `turning column indexes off`_ will always improve
+performance. Sometimes significantly. But the downside is that you cannot use
+those columns in the where clause.
+
+Primary Keys
+------------
+
+If your data does not have a `natural primary key`_ (i.e. data that uniquely
+identifies each row), use the ``_id`` `system column`_ as a primary key. This
+is better than creating your own `surrogate primary key`_ (e.g. manually
+generating a UUID for each row) because there is one less column to index.
+
+Translog
+--------
+
+If `translog.durability`_ is set to ``REQUEST`` (the default), the
+:ref:`translog <durability>` gets flushed after every operation. Setting this
+to ``ASYNC`` will improve insert performance, but it also worsens durability.
+If a node is shut down before a translog has been synced, those opperations
+will be lost.
+
+Refresh Interval
+----------------
+
+Data that has been written to a shard cannot be read back until the shard index
```

Comment:
With primary key lookups it is possible to retrieve written data immediately without explicit refresh. - Maybe re-phrase this slightly to take that into account?

Reply:
changed to: 

> With the exception of primary key lookups, data that has been written to a
shard cannot be read back until the shard index has been refreshed.